### PR TITLE
fix(draft): Add distinctUntilChanged for auto save

### PR DESCRIPTION
**Pull request by antoniskalou**
This should reduce the likelyhood that a draft will be saved after it has been sent.

See https://github.com/runbox/runbox7/issues/644

### CODE REVIEW

**Review by perigrin — APPROVED** [review]
This looks reasonable now that I've looked at the docs for `distinctUntilChanged()`.

**Comment by castaway** [comment]
Hmm, dUC uses "===" by default, is that doing to dwim? https://www.learnrxjs.io/learn-rxjs/operators/filtering/distinctuntilchanged 

Might need to check the fields, eg like https://github.com/runbox/runbox7/blob/master/src/app/rmmapi/messagelist.service.ts#L79

File: src/app/common/util.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// --------- BEGIN RUNBOX LICENSE ---------
+// Copyright (C) 2016-2018 Runbox Solutions AS (runbox.com).
```
Comment (castaway) [inline src/app/common/util.ts:2]: 2023!

File: src/app/common/util.spec.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+// --------- BEGIN RUNBOX LICENSE ---------
+// Copyright (C) 2016-2018 Runbox Solutions AS (runbox.com).
```
Comment (castaway) [inline src/app/common/util.spec.ts:2]: 2023!

File: src/app/common/util.spec.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+// General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with Runbox 7. If not, see <https://www.gnu.org/licenses/>.
+// ---------- END RUNBOX LICENSE ----------
+
+import { objectEqual, withKeys, objectEqualWithKeys } from "./util";
+
+describe('util', () => {
+    it('objectEqual', () => {
```
Comment (castaway) [inline src/app/common/util.spec.ts:23]: Slightly more descriptive test names please?

**Review by castaway — CHANGES_REQUESTED** [review]
Couple of minor updates please ;)

**Review by castaway — APPROVED** [review]
lgtm

**Pull request merged**